testserver/testserver_test.go

```diff
@@ -234,6 +234,12 @@ func TestRunServer(t *testing.T) {
 				)
 			},
 		},
+		{
+			name: "No File Cleanup",
+			instantiation: func(t *testing.T) (*sql.DB, func()) {
```

Comment:
is there any way for this test to assert that the directory still exists at the end?

Reply:
Added a dedicated test `TestBaseDirIsPreservedWhenNoFileCleanupRequested` (but also had to add some additional machinery to power it).